graphql/resolvers.go

```diff
@@ -441,6 +444,131 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
 	return groupedFilesList, nil
 }
 
+func (r *queryResolver) TaskLogs(ctx context.Context, taskID string) (*RecentTaskLogs, error) {
```

Comment:
this function is realllll long which usually entails that the function is comprised of multiple sets of logic all in the same function body. i look at functions like this the same way i look at really big react components. i wonder "can this component be broken into smaller chunks that make reading the component much simpler"? what hels me a lot when reading large function bodies is when it reads like an index of the logic...

```
func (r *queryResolver) TaskLogs(ctx context.Context, taskID string) (*RecentTaskLogs, error) {
  tasks, err := getTaskLogsAndReorderThem()
  filteredTasks, err := doSomethingWithTasks(tasks)
  endResult, err := doMoreThings(tasks)

  return endResult, nil
}
```

for example, this first part of the function looks like it fetches the logs and then performs some logic around removing scheduled events and then re-ordering the list. would this chunk of logic makes more sense in its own func?

just some food for thought as the resolver file grows larger and larger :)

Reply:
I agree this function is fairly long. I think it makes sense to create functions from code could be reusable like looping items to append their pointers. I added some comments around the other logic to make it more readable.

Reply:
code does need to be reusable to warrant it being in a function. functions serve purpose of encapsulating logic regardless of if it's reused.